dwave/optimization/symbols.pyx

```diff
@@ -781,6 +783,12 @@ cdef class Constant(ArraySymbol):
         # Have the parent model hold a reference to the array, so it's kept alive
         model._data_sources.append(array)
 
+    def __bool__(self):
+        if not self._is_scalar():
+            raise ValueError("the truth value of a constant with more than one element is ambiguous")
+
+        return <bool>deref(self.ptr.buff())
```

Comment:
Does the `buff()` method point at the first (only in this case) location in memory? What is the difference between what's returned by `self.ptr` (I assume this is a pythonic pointer object and not actually a memory address pointer?) and `self.ptr.buff()`?